[workspace] the great upgrade

codingpeanut157 opened the pull request:
### **User description**
this commit upgrades:
- bcr-common to v0.4.0 (yet yo be release)
- bcr-ebill-core v0.4.10 (will be upgraded again once aligned to latest bcr-common)
- cashu to v0.13

it also removes bcr-wdc-quote-client and replaces it with bcr-common::client::quote


___

### **PR Type**
Enhancement, Refactoring


___

### **Description**
**Major Dependency Upgrades:**

- Upgraded `bcr-common` to v0.4.0 with comprehensive integration across all services

- Upgraded `bcr-ebill-core` to v0.4.10

- Upgraded `cashu` to v0.13

- Removed `bcr-wdc-quote-client` crate and replaced with `bcr-common::client::quote`

**Type Safety Improvements:**

- Introduced strongly-typed domain objects (`Address`, `City`, `Country`, `Email`, `Name`, `Zip`, `Date`) replacing plain strings

- Replaced `Uuid` with `PaymentIdentifier` type throughout payment flows

- Changed URL types from `String` to `url::Url` for better type safety

- Updated conversion functions to return `Result` types for proper error handling

**Architecture Refactoring:**

- Refactored `Service` struct to use `Arc<dyn Trait>` instead of generic type parameters for dependency injection

- Simplified wallet implementation by using trait objects (`Box<dyn PrivateKeysRepository>`)

- Removed generic parameters from production types and endpoint handlers

**Payment System Enhancements:**

- Added `deadline` parameter support across payment request flows

- Added `reserved_fees` field to outgoing payment requests

- Updated payment methods with new `IncomingPaymentOptions` and `OutgoingPaymentOptions` structures

- Added `wait_payment_event` and enhanced `check_incoming_payment_status` methods

**Wire Protocol Migration:**

- Migrated all services to use `bcr_common::wire` modules (`wire::quotes`, `wire::bill`)

- Updated client implementations to use `bcr_common::client` modules (keys, quote, swap)

- Standardized error types across services to use bcr-common error types

**Configuration & Infrastructure:**

- Added `CourtClient` integration with `CourtConfig` and `DevModeConfig`

- Removed deterministic key derivation in favor of random secret generation

- Added explicit denomination amounts to keyset generation

- Changed seed parameter type from `&[u8]` to `[u8; 64]` for consistency

**Testing:**

- Commented out unit tests in payment service (marked for future updates)

- Updated test utilities to use new type-safe domain models

- Migrated E2E test clients to bcr-common modules


___

### Diagram Walkthrough


```mermaid
flowchart LR
  A["bcr-wdc-quote-client"] -- "removed" --> B["bcr-common::client::quote"]
  C["String types"] -- "replaced with" --> D["Domain objects<br/>(Address, Email, Date)"]
  E["Uuid"] -- "replaced with" --> F["PaymentIdentifier"]
  G["Generic parameters"] -- "simplified to" --> H["Arc<dyn Trait>"]
  I["bcr-wdc-webapi types"] -- "migrated to" --> J["bcr-common::wire"]
  K["Payment API"] -- "enhanced with" --> L["deadline & reserved_fees"]
```



<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>14 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>service.rs</strong><dd><code>Refactor payment service to use trait objects and update payment API</code></dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/service.rs

<ul><li>Updated trait bounds to require <code>Send</code> for <code>OnChainWallet</code>, <br><code>PaymentRepository</code>, and <code>EBillNode</code> traits<br> <li> Refactored <code>Service</code> struct to use <code>Arc<dyn Trait></code> instead of generic <br>type parameters for dependency injection<br> <li> Modified payment methods to use new <code>PaymentIdentifier</code> type instead of <br><code>Uuid</code> and updated payment flow with new <code>IncomingPaymentOptions</code> and <br><code>OutgoingPaymentOptions</code> structures<br> <li> Updated <code>MintPayment</code> trait implementation with new method signatures <br>including <code>wait_payment_event</code>, <code>check_incoming_payment_status</code> returning <br><code>Vec<WaitPaymentResponse></code>, and added <code>deadline</code> parameter to <code>request_to_pay</code><br> <li> Commented out all unit tests (marked for future updates)</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-b5acbddaada54279fd8bdaa1fd26b270a4cd28251ced566869426ee85da7c965">+410/-430</a></td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>convert.rs</strong><dd><code>Introduce type-safe domain models and enhance conversion functions</code></dd></summary>
<hr>

crates/bcr-wdc-utils/src/convert.rs

<ul><li>Added new strongly-typed domain objects (<code>Address</code>, <code>City</code>, <code>Country</code>, <br><code>Email</code>, <code>Name</code>, <code>Zip</code>, <code>Date</code>) replacing plain strings in conversion <br>functions<br> <li> Updated conversion functions to return <code>Result</code> types for error handling <br>during validation<br> <li> Added new conversion functions for <code>BillToShareWithExternalParty</code> and <br>payment-related structures<br> <li> Enhanced bill data structures with additional fields like <br><code>endorsements</code>, <code>payment_deadline_timestamp</code>, and <code>mint</code> status</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-52321ba996233054c565727cc2472f705539d2c40e36fbed7f02fcbda1a2cf3a">+209/-125</a></td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>surreal.rs</strong><dd><code>Update payment persistence layer with new identifier type</code></dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/persistence/surreal.rs

<ul><li>Replaced <code>Uuid</code> with <code>PaymentIdentifier</code> type for payment request <br>identification<br> <li> Updated database record ID generation to use <br><code>PaymentIdentifier::to_string()</code><br> <li> Added <code>reserved_fees</code> field to <code>OutgoingPaymentDBEntry</code> structure<br> <li> Modified all payment repository methods to work with new identifier <br>type</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-154a8606225da6382c0acc6a45b41077e152771dbc49692e496c2183b3eee96f">+41/-30</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>onchain.rs</strong><dd><code>Simplify wallet implementation with trait objects</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/onchain.rs

<ul><li>Changed <code>PrivateKeysRepository</code> trait to require <code>Send + Sync + Debug</code> <br>bounds<br> <li> Refactored <code>Wallet</code> struct to use <code>Box<dyn PrivateKeysRepository></code> instead <br>of generic type parameter<br> <li> Removed manual <code>Clone</code> implementation and updated wallet synchronization <br>to use reference instead of <code>Arc</code><br> <li> Simplified <code>SignOptions</code> usage by using default values directly</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-387525c9e9cfc8ee582e74e6e833885c3f990618b92bac2f0ca0727e9a3a18e5">+22/-46</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>web.rs</strong><dd><code>Integrate type-safe domain models in web API endpoints</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebill-service/src/web.rs

<ul><li>Added imports for new domain types (<code>City</code>, <code>Country</code>, <code>Date</code>, <code>Email</code>, <code>Name</code>, <br><code>Identification</code>)<br> <li> Updated bill-related endpoints to use <code>wire::quotes</code> and <code>wire::bill</code> <br>types from <code>bcr-common</code><br> <li> Modified <code>create_identity</code> to validate and construct strongly-typed <br>domain objects<br> <li> Added <code>deadline</code> parameter to <code>request_to_pay_bill</code> endpoint and updated <br>bill action execution</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-d8457ecbd57dbb19b898ff61af1dcc6112b91759fc270f19d2a05fa479c5ebcc">+50/-30</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>lib.rs</strong><dd><code>Add court client and dev mode configuration support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebill-service/src/lib.rs

<ul><li>Added <code>CourtConfig</code> and <code>DevModeConfig</code> to <code>AppConfig</code> structure<br> <li> Changed URL types from <code>String</code> to <code>url::Url</code> for better type safety<br> <li> Added <code>CourtClient</code> initialization and integration into bill service<br> <li> Updated service constructors to include new court and notification <br>dependencies</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-bca71cbba9ed5622ab5e4c3fa1910a48a033a88332b9531db05bd4869694b8ce">+56/-20</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>mod.rs</strong><dd><code>Remove deterministic key derivation in favor of random secrets</code></dd></summary>
<hr>

crates/bcr-wdc-treasury-service/src/credit/mod.rs

<ul><li>Removed <code>xpriv</code> field from <code>Service</code> struct<br> <li> Removed counter-related methods from <code>Repository</code> trait<br> <li> Changed premint secret generation from <code>from_xpriv</code> to <code>random</code> method<br> <li> Removed counter increment logic from quote creation flow</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-fb8e907942c08710bc61e846f07cd964a3a620ff03e6df23ec32a299dc7a6558">+7/-26</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>factory.rs</strong><dd><code>Add explicit denomination amounts to keyset generation</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-key-service/src/factory.rs

<ul><li>Added <code>denominations</code> vector calculation for keyset generation<br> <li> Updated <code>MintKeySet::generate</code> call to accept denominations slice<br> <li> Added <code>amounts</code> field to <code>MintKeySetInfo</code> with calculated denominations</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-0936b33fd56c8a54b5936db31eea011621bb1d417e92c43b5751a1bac4ed393a">+4/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>ebill.rs</strong><dd><code>Add deadline parameter to bill payment requests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/ebill.rs

<ul><li>Updated imports to use <code>bcr_common::core::BillId</code> and <br><code>bcr_common::wire::bill</code><br> <li> Added <code>deadline</code> parameter to <code>request_to_pay</code> method signature<br> <li> Updated <code>RequestToPayBitcreditBillPayload</code> to include <code>deadline</code> field</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-8739b18a417b2716009f8331ab463bc22173635dd226ba9394cae154128c342a">+16/-8</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>keys.rs</strong><dd><code>Update keyset generation to use explicit denominations</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-utils/src/keys.rs

<ul><li>Added <code>denominations</code> vector calculation in keyset generation functions<br> <li> Updated <code>MintKeySet::generate_from_seed</code> calls to accept denominations <br>slice<br> <li> Added <code>amounts</code> field to <code>MintKeySetInfo</code> structures in test utilities</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-88918b1bb74f41ff3324615d303d1c32ed912a92eeefe25d47604cf2b3a362c8">+6/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>surreal.rs</strong><dd><code>Add denominations field to keyset database entry</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-key-service/src/persistence/surreal.rs

<ul><li>Added <code>denominations</code> field to <code>KeysInfoDBEntry</code> struct<br> <li> Updated conversion implementations to include <code>amounts</code> field mapping</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-ba8ade7df8f929ce375f54a312f93f59224a2beafd836727aa2108c57d32ca26">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>payment.rs</strong><dd><code>Add reserved fees tracking to outgoing payment requests</code>&nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/payment.rs

<ul><li>Changed request ID type from <code>Uuid</code> to <code>PaymentIdentifier</code><br> <li> Added <code>reserved_fees</code> field to <code>OutgoingRequest</code> struct<br> <li> Updated <code>OutgoingRequest::new</code> to accept and store <code>reserved_fees</code> <br>parameter</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-4c8f4336286dc55b7b8a0c9733e4d67f73a6a91ef737b61d48728f2b4104c519">+10/-4</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>service.rs</strong><dd><code>Add deadline support to ebill minting requests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-treasury-service/src/debit/service.rs

<ul><li>Added <code>deadline</code> parameter to <code>mint_from_ebill</code> method signature<br> <li> Updated imports to use <code>bcr_common::core::BillId</code> and added <code>TStamp</code> type<br> <li> Included <code>deadline</code> field in <code>RequestToMintFromEBillDesc</code> structure</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-caaa9a1592f0e5ce710ca52d8f651b10978b6986cb1090c391f76ab4bd749b06">+8/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>admin.rs</strong><dd><code>Add deadline parameter to mint from ebill endpoint</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-treasury-service/src/admin.rs

<ul><li>Added <code>deadline</code> parameter extraction from request<br> <li> Passed <code>deadline</code> to <code>mint_from_ebill</code> method call</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-1a696476126e3290a8fea9f62843118cdb79f77c324ead24a867267d7dacaff3">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>admin.rs</strong><dd><code>Migrate quote admin endpoints to use bcr-common wire types</code></dd></summary>
<hr>

crates/bcr-wdc-quote-service/src/admin.rs

<ul><li>Updated imports to use <code>wire::quotes</code> from <code>bcr-common</code> instead of <br><code>bcr-wdc-webapi</code><br> <li> Modified conversion functions to handle new <code>Result</code> return types from <br>bill conversion utilities<br> <li> Updated API endpoint handlers to work with new wire protocol types</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-0ac7a81b04baaf01768d8526f8dceb0754d20adf93d53808f189a4b810752acb">+64/-63</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>service.rs</strong><dd><code>Migrate quote service to bcr-common wire protocol types</code>&nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-quote-service/src/service.rs

<ul><li>Updated imports to use <code>BillId</code> and <code>NodeId</code> from <code>bcr-common::core</code> and <br><code>wire::quotes</code> types<br> <li> Modified <code>EBillNode</code> trait method signature to use <br><code>wire_quotes::SharedBill</code> and <code>wire_quotes::BillInfo</code><br> <li> Updated quote service methods to work with new wire protocol types<br> <li> Refactored test utilities to use <code>wire_tests</code> and <code>core_tests</code> from <br><code>bcr-common</code></ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-186f7e8fe93bbc7070dc420fa5c734f262aa91e2cc4a5585eecc838222112f36">+27/-88</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>test_utils.rs</strong><dd><code>Refactor test utilities with type-safe domain models</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-webapi/src/test_utils.rs

<ul><li>Refactored test utility functions to use strongly-typed domain objects <br>(<code>Country</code>, <code>City</code>, <code>Address</code>, <code>Date</code>) instead of strings<br> <li> Removed <code>random_identity_public_data</code> and related functions (moved to <br><code>wire_tests</code>)<br> <li> Updated <code>generate_random_bill_enquire_request</code> to use new conversion <br>functions with error handling<br> <li> Simplified <code>random_date</code> function to return <code>Date</code> type instead of string</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-e1f74ab6df7fe46ff507cbce007fc16efe9c4c63181b4f6025bea28d976819f3">+27/-189</a></td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_utils.rs</strong><dd><code>Update test utilities to use bcr-common wire types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-e2e-tests/src/test_utils.rs

<ul><li>Updated import to use <code>bcr_common::wire::quotes as wire_quotes</code><br> <li> Changed <code>bill</code> field type in <code>EbillRequestComponents</code> to use <br><code>wire_quotes::SharedBill</code></ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-22f15da9c72f8af83a1e2828f15e27fc283744d8f8a21d03cdbab3d9d82f75ce">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Refactoring</strong></td><td><details><summary>20 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>web.rs</strong><dd><code>Migrate quote web API types to bcr-common wire module</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-quote-service/src/web.rs

<ul><li>Replaced <code>bcr_wdc_webapi::quotes</code> import with <code>bcr_common::wire::quotes </code><br><code>as wire_quotes</code><br> <li> Updated all type references from <code>web_quotes::*</code> to <code>wire_quotes::*</code> <br>throughout the file<br> <li> Modified function signatures and return types to use the new <br><code>wire_quotes</code> namespace</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-1bd0e6958ea77d8d30de256aa68ee557a88bc615c71586889a98edca3ea18e30">+24/-24</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>lib.rs</strong><dd><code>Refactor ebill client to use bcr-common types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebill-client/src/lib.rs

<ul><li>Updated imports to use <code>bcr_common::core::BillId</code> and <code>bcr_common::wire</code> <br>modules<br> <li> Removed <code>bcr_wdc_webapi</code> imports and replaced with <code>bcr_common</code> <br>equivalents<br> <li> Changed type references from <code>bcr_wdc_webapi</code> types to <code>wire_bill</code> and <br><code>wire_quotes</code> types<br> <li> Added <code>#[allow(dead_code)]</code> attributes to unused fields</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-a16ec5eb3940ebaff1e6279b5b37f628510942532a0e1720c6b5617e0ad787dd">+21/-16</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>quotes.rs</strong><dd><code>Change maturity date representation and improve type conversions</code></dd></summary>
<hr>

crates/bcr-wdc-quote-service/src/quotes.rs

<ul><li>Changed <code>maturity_date</code> type from <code>TStamp</code> to <code>chrono::NaiveDate</code><br> <li> Updated imports to use <code>bcr_common::wire::quotes</code> instead of <br><code>bcr_wdc_webapi::quotes</code><br> <li> Changed <code>From</code> implementation to <code>TryFrom</code> for better error handling<br> <li> Updated conversion functions to handle new date type and return <code>Result</code></ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-03545ff0e2dffaef5dbb0e2990ee74cf84c03a3c10f8f0f069b28ac5734f1ef8">+23/-26</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>signatures.rs</strong><dd><code>Simplify cashu type imports and remove aliasing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-utils/src/signatures.rs

<ul><li>Updated imports to use <code>cashu</code> types directly instead of aliased <code>cdk00</code>, <br><code>cdk01</code>, <code>cdk02</code><br> <li> Changed function signatures to use <code>cashu::BlindedMessage</code> and <br><code>cashu::Proof</code> types<br> <li> Updated test utilities to use <code>cashu::MintKeySet</code> and related types</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-3a6b632ae0793c851aa4bac969ce483cc94c1bdc2954c1af74d159ae2de91e8f">+14/-15</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>inmemory.rs</strong><dd><code>Replace UUID with PaymentIdentifier for payment requests</code>&nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/persistence/inmemory.rs

<ul><li>Changed payment request ID type from <code>Uuid</code> to <code>PaymentIdentifier</code><br> <li> Updated HashMap key types and method signatures to use <br><code>PaymentIdentifier</code><br> <li> Modified store operations to clone <code>PaymentIdentifier</code> when inserting</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-b7d646c115310b441dc975285b443b008bd7705816c04aed78cf3468cae189ab">+10/-10</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>ebill.rs</strong><dd><code>Migrate ebill client to bcr-common wire types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-quote-service/src/ebill.rs

<ul><li>Updated imports to use <code>bcr_common::wire::quotes</code> instead of <br><code>bcr_wdc_webapi::quotes</code><br> <li> Changed <code>base_url</code> type from custom <code>Url</code> to <code>reqwest::Url</code><br> <li> Updated test utilities to use new date generation and string parsing</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-8f49a9e1d88ad9c59e7ebeb22dd015f30ef99d14528d174bdbe643e56161c81d">+13/-14</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>clients.rs</strong><dd><code>Migrate E2E test clients to bcr-common client modules</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-e2e-tests/src/clients.rs

<ul><li>Replaced <code>bcr_wdc_quote_client::QuoteClient</code> with <br><code>bcr_common::client::quote::Client</code><br> <li> Updated imports to use <code>bcr_common::client</code> modules for keys, quote, and <br>swap clients<br> <li> Changed type references from <code>bcr_wdc_webapi::quotes</code> to <br><code>bcr_common::wire::quotes</code></ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-e590f662f5fd8bbe6a53eebe1a3fc173fd9192bab5fe99d9f23caecf0540ebd1">+13/-10</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>wildcat.rs</strong><dd><code>Refactor wildcat client to use bcr-common modules</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-treasury-service/src/debit/wildcat.rs

<ul><li>Replaced <code>bcr_wdc_quote_client</code> with <code>bcr_common::client::quote::Client</code><br> <li> Updated imports to use <code>bcr_common::client</code> modules and <br><code>bcr_common::core::BillId</code><br> <li> Changed <code>quote_service_url</code> type from custom type to <code>reqwest::Url</code><br> <li> Updated status discriminant references to use <code>wire_quotes</code> namespace</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-b00e7a42e7f7d62211a82197682b868526c0e095d5488b58a77be62a99abf922">+19/-13</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>lib.rs</strong><dd><code>Simplify production types by removing generic parameters</code>&nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/lib.rs

<ul><li>Changed <code>ProdOnChainWallet</code> type to remove generic parameters<br> <li> Updated <code>ProdService</code> type definition to remove generic parameters<br> <li> Modified wallet initialization to use <code>Box<dyn></code> for key repository<br> <li> Changed service constructor to use <code>Arc</code> wrappers for dependencies</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-4a6232be2624d5874e7767c11f23f5f0d870347f5291203bf0de0f4d7d6f7b9d">+11/-9</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>wallet.rs</strong><dd><code>Update wallet initialization and mint info fetching methods</code></dd></summary>
<hr>

crates/bcr-wdc-treasury-service/src/debit/wallet.rs

<ul><li>Changed <code>seed</code> parameter type from <code>&[u8]</code> to <code>[u8; 64]</code> in <code>CDKWallet::new</code><br> <li> Replaced <code>get_mint_info</code> call with <code>fetch_mint_info</code><br> <li> Updated <code>keysets_info</code> to use <code>get_mint_keysets</code> instead of <br><code>get_active_mint_keysets</code></ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-ab70a3c30b817240c79e6db77038f6efecb7530eb01f06331c389f4440f08440">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>error.rs</strong><dd><code>Update bcr-common error type paths</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-wallet-aggregator/src/error.rs

<ul><li>Updated error type paths from <code>bcr_common::KeysError</code> to <br><code>bcr_common::client::keys::Error</code><br> <li> Updated error type paths from <code>bcr_common::SwapError</code> to <br><code>bcr_common::client::swap::Error</code></ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-83abe012df823c2cf0711368e006172e98a578f9226eebceba77a51cdecbd7d6">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>lib.rs</strong><dd><code>Update wallet aggregator to use bcr-common client modules</code></dd></summary>
<hr>

crates/bcr-wdc-wallet-aggregator/src/lib.rs

<ul><li>Updated client type paths to use <code>bcr_common::client::keys::Client</code> and <br><code>bcr_common::client::swap::Client</code><br> <li> Changed client initialization to use new module paths</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-8479e17ae2f35c019c884788163137e22c66a8955874a28a3ebbe30754be3c1c">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>admin.rs</strong><dd><code>Remove generic parameters from admin balance endpoint</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/admin.rs

<ul><li>Removed generic type parameters from <code>balance</code> function signature<br> <li> Simplified function to use concrete <code>Service</code> type</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-77f7de24a67f1ff71dee268db39ea6fa46b1e51ef0d01f2b2f635d398b0c8226">+2/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>web.rs</strong><dd><code>Remove generic parameters from network endpoint</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/web.rs

<ul><li>Removed generic type parameters from <code>network</code> function signature<br> <li> Simplified function to use concrete <code>Service</code> type</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-25fedaba14cb8407c2ab010f615e7d5bd0f5fd9dc2a8a8aad25a1ca9dfeabafe">+2/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>devtool.rs</strong><dd><code>Update devtool to use bcr-common wire types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-webapi/src/bin/devtool.rs

<ul><li>Added import for <code>bcr_common::wire::quotes as wire_quotes</code><br> <li> Updated <code>SignedEnquireRequest</code> type reference to use <code>wire_quotes</code> <br>namespace</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-df21ec837a02726f23714c89a58bfc9afeb0f0e568a977a8907f7bed7f32c477">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>error.rs</strong><dd><code>Update payment error to use PaymentIdentifier type</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebpp/src/error.rs

<ul><li>Changed <code>PaymentRequestNotFound</code> error variant to use <code>PaymentIdentifier</code> <br>instead of <code>Uuid</code><br> <li> Removed <code>uuid::Uuid</code> import</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-40f09e3a0057521b4a777339a6cf9a9550d27a721d0260214e412fa1c0fdc521">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>lib.rs</strong><dd><code>Update seed parameter type in treasury controller</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-treasury-service/src/lib.rs

<ul><li>Changed <code>seed</code> parameter type from <code>&[u8]</code> to <code>[u8; 64]</code> in <br><code>AppController::new</code></ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-c37ae0172b7c48d3ef7ff2f2b24ff060ac7f261babf3b0a7c21720bde394c041">+1/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>inmemory.rs</strong><dd><code>Simplify maturity date filtering logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-quote-service/src/persistence/inmemory.rs

<ul><li>Changed maturity date comparison from <code>date_naive()</code> method to direct <br>comparison with <code>chrono::NaiveDate</code></ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-ee1777cd285c0ad54e3025267b8ca35f5ef55ff80cb8d79bc3b98c4984d8bfd3">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>main.rs</strong><dd><code>Update seed type in treasury service main</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-treasury-service/src/main.rs

<ul><li>Changed seed array type from slice reference to fixed-size array <code>[u8; </code><br><code>64]</code> in controller initialization</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-ce3b84223bbd3d0908da85d994cb4a235eb6f85bed167479654f1f30c2b4268e">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>error.rs</strong><dd><code>Update keys client error type path</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-swap-service/src/error.rs

<ul><li>Updated <code>KeysClient</code> error type from <code>bcr_common::KeysError</code> to <br><code>bcr_common::client::keys::Error</code></ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-1efd0eb1b6134bb669567aad1c0684679f39c90daef1e35a29c11cb82bf63433">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Error handling</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>error.rs</strong><dd><code>Expand error handling for service and validation errors</code>&nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebill-service/src/error.rs

<ul><li>Added new error variants to <code>ServiceError</code> response mapping<br> <li> Extended <code>ValidationError</code> response handling with additional validation <br>error types</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-b935147390b93205af89a3d7114a4cb905a2615eb1ef9cc7604d599ec4c2bcfd">+11/-5</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>error.rs</strong><dd><code>Add conversion error handling and update keys error type</code>&nbsp; </dd></summary>
<hr>

crates/bcr-wdc-quote-service/src/error.rs

<ul><li>Added <code>Convert</code> error variant wrapping <code>bcr_wdc_utils::convert::Error</code><br> <li> Updated <code>KeysHandler</code> error type from <code>bcr_common::KeysError</code> to <br><code>bcr_common::client::keys::Error</code><br> <li> Added <code>Convert</code> error handling in response mapping</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-ba4223300771963ac9bac2c3a6d4859b9e260cff57010253b4ef8f8d51f641ef">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>bill.rs</strong><dd><code>Improve error handling in bill conversion</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-webapi/src/bill.rs

<ul><li>Updated conversion functions to handle <code>Result</code> types with <code>?</code> operator<br> <li> Added <code>transpose()</code> call for optional field conversion</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-6a495b86b4d732551ef6dad860fb79491bf2762aa3037122a4a84cebe72c0e03">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>surreal.rs</strong><dd><code>Remove counters configuration from credit connection config</code></dd></summary>
<hr>

crates/bcr-wdc-treasury-service/src/persistence/surreal.rs

- Removed `counters` field from `CreditConnectionConfig` struct


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-a039943f71444ec616e3d73977f35b1e3e55235241fc11c0e9151c739621abfd">+2/-54</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>main.rs</strong><dd><code>Initialize court and dev mode configurations in main</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

crates/bcr-wdc-ebill-service/src/main.rs

<ul><li>Added <code>CourtConfig</code> and <code>DevModeConfig</code> imports and initialization<br> <li> Updated API config construction to include court and dev mode <br>configurations</ul>


</details>


  </td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-c5c3ed2056e794256111141fb0aae3454eadb997ce4f32672f76e404d8c19c9e">+7/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>28 files</summary><table>
<tr>
  <td><strong>Cargo.toml</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542">+12/-12</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>Cargo.toml</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-5aa2b35cb73a0e4968d3853eed4415b5acef0c7382b97804d16eff5e024e1955">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>main.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-7dfaf4fa6e45c9464ca129bab5799f4ef75cb2e6e8bb38769a49b510506f6a86">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Cargo.toml</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-fbb36d8aa1db230a56172e02eeb7630c6b0ff44f9c10f92d56ad92668e9fdbeb">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Cargo.toml</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-3fe552818ead67e1b09da15ad6b91494204ec912c55ff6335bf7fe4d30675c9d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>client.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-8b5dd0c09ffad825927b2e33578a23b31c031f59fb0de8ca030e1d9fad8b42c2">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>lib.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-3f8809d40f9e828f899b297d0eed35852796e26fa96f3c629ac0f4a6dc079589">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>keyset_info.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-a2e73de7c98e5bc21edda9e3af565f55e425344589e8d32b5d1c0929358e0d26">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>list_keys.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-ca43e23d1bc46c4a95f85a9c767f990183f8f4c10425d6d2e0c4131243f922a3">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>list_keyset_info.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-0be850109b92666253ef2ebbaa20947be0121f86c4168565dffad60f00bd1b77">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>restore.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-341dd45df3eef513f9edbed9320703ab5fde6adb1bd8a8ad15f5c5b41281c4aa">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Cargo.toml</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-4e256eed7ead076418f608369ce678e64c1a2ee653ee3518b1f27a22b3c502b0">+0/-26</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>lib.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-fa0779382859a62530dd708ed21f321792e966dee691bb62cb62946804e28a43">+0/-235</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>cancel.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-8d894fc4076b3af42631c49af8068695d9d222436fffbc73a3ac5058a8d7b3de">+0/-17</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>enquire.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-c1aa452a491e88b5931fae2864857a2a86487522824636ca1e8ca0c8cbe1ffd6">+0/-24</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>list.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-03e6c0af2a4326a20b8eabba92723a19a6b632f06a408607af0c7a1d1d8c3058">+0/-69</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>lookup.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-d1f56eaa033a97db8a88e1fcc6320a8fce933dcbb21105719396a0f567f32082">+0/-49</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>keys.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-d24b03325dacb5f2241120b3f10ef53990566604d941fe8a1dfd7a4ee08098ed">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>surreal.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-e0caddd16b02b3646a69fca2fb0c9f351fcb57dcd7d613a1c750a4e1bd051a0b">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>keys.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-ee737c4d0352b0fcb329c16d410f63ab5f29e139ec20200c8440f4ca763052b4">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>lib.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-d6043a0bd0e59231f714995c9050c95248343410c6e4944cd2d027c37ded0b06">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>checkstate.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-d6f615157c6972f07e64509cf49836b3ec416e71b2cb84a7f3a5dd8ed44cb675">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Cargo.toml</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-6df636339e3a8a42ba9e410dd9560703ec90157f097e903f0e733f8b4d18f4cc">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>keys.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-9add08abfe7f7f8e016f41d9db0813e65951b9f1db294c4229e7e29e5a78fa78">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>error.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-ab2c84ccec5c1e98a9c43fc0ab1e435af80833007aa3e33f25260f5397da2214">+0/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>inmemory.rs</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-e198da7e8e5ddc23f04277ff4c9b8cdbd33244c722021b154c1ceb0bc8a8e3ab">+0/-19</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Cargo.toml</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-244fb9160ddcbeadb2c1faa1830722c935612662a349aa529b15589c440f4e6b">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Cargo.toml</strong></td>
  <td><a href="https://github.com/BitcreditProtocol/Wildcat/pull/317/files#diff-245d40508bcd3a2d2446ceb2a51102995a7dab7aeabdd2b68e1de9528a4e4d1d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

zupzup commented on crates/bcr-wdc-ebpp/src/ebill.rs:
```diff
@@ -27,15 +29,21 @@ impl EBillClient {
+        &self,
+        bill: &BillId,
+        amount: Amount,
+        deadline: TStamp,
```
just as a note - this deadline currently has to be at least `end-of-day(now() + 48h)` to be valid

codingpeanut157 replied:
I assume the ebill-service is going to reject any incorrect value, rejecting the entire request-to-pay

zupzup replied:
yes :+1:

zupzup commented on crates/bcr-wdc-ebpp/src/onchain.rs:
```diff
@@ -40,54 +40,33 @@ pub struct WalletConfig {
-    // configs
-    network: Network,
+    repo: Box<dyn PrivateKeysRepository>,
+    electrum_client: Arc<BdkElectrumClient<ElectrumApi>>, // configs network: Network,
```
I think the comment at the end is a leftover

zupzup commented on crates/bcr-wdc-treasury-service/src/admin.rs:
```diff
@@ -79,7 +79,9 @@ where
     //TODO! wait for bitcredit-core to integrate bcr-common
     let ebill_id = bcr_ebill_core::bill::BillId::from_str(&request.ebill_id.to_string())
```
this can be updated as well now, right?

zupzup commented on crates/bcr-wdc-ebpp/src/service.rs:
```diff
@@ -175,7 +170,10 @@ where
                 let ebill_id =
```
why is this `expect` instead of error-handled?

And shouldn't this also use bcr-common's billId?

zupzup commented on crates/bcr-wdc-ebpp/src/service.rs:
```diff
@@ -500,300 +480,300 @@ fn parse_to_bip21_uri(input: &str, network: btc::Network) -> Result<bip21::Uri<'
-        assert!(result.is_err());
-        assert!(matches!(result.unwrap_err(), PaymentError::Amount(_)));
-    }
+    // use super::*;
```
why all commented out? 😅 

Is this basically a "TODO(future): fix tests"?

codingpeanut157 replied:
yeah sorry
this PR is a bit messy

I am waiting for the next release of bitcredit-core to resolve conflicting versions of bcr-common

zupzup reviewed (approved):
LGTM 👍

stefanbitcr commented on crates/bcr-wdc-key-service/src/factory.rs:
```diff
@@ -50,15 +49,17 @@ impl Factory {
             .master
             .derive_priv(secp, &path)
             .expect("bitcoin::derive_priv unexpected error");
+        let denominations: Vec<u64> = (0..Self::MAX_ORDER).map(|i| 2_u64.pow(i as u32)).collect();
```
Unrelated to this PR, but looks like new Cashu will refactor the amounts out later into an explicit FeesAndAmounts anyway

stefanbitcr reviewed (approved):
Awesome

codingpeanut157 committed:
[workspace] the great upgrade

this commit upgrades:
- bcr-common to v0.4.0 (yet yo be release)
- bcr-ebill-core v0.4.10 (will be upgraded again once aligned to latest
  bcr-common)
- cashu to v0.13

it also removes bcr-wdc-quote-client and bcr-wdc-ebill-client and
replaces it with equivalents from bcr-common::clients
